### PR TITLE
New Providers

### PR DESCRIPTION
**Pull request by KillerDogeEmpire**
added MtlNovel, and Efrem.net with the help of Addipe

Efrem.net is a clone of readfromnet, however, when readfromnet gets updated the code breaks, but efrem.net doesn't break, since it doesn't get the same updates as readfromnet even though it's a clone

### CODE REVIEW

**Comment by KillerDogeEmpire** [comment]
I also changed the load response to the new StreamResponse

File: app/src/main/java/com/lagradost/quicknovel/providers/EfremnetProvider.kt [hunk 1]
```diff
@@ -0,0 +1,1502 @@
...
+        data.add(
+            ChapterData(
+                "page 1",
+                url.substringBeforeLast("/") + "/page,1," + url.substringAfterLast("/"),
+                null,
+                null
+            )
+        )
+
+        for (c in 0..(chapters.size / 2)) {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/quicknovel/providers/EfremnetProvider.kt:1471]: any reason why /2

File: app/src/main/java/com/lagradost/quicknovel/providers/MtlNovelsProvider.kt [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+    override suspend fun search(query: String): List<SearchResponse> {
+        val response =
+            SearchResults.fromJson(app.get("https://www.mtlnovel.com/wp-admin/admin-ajax.php?action=autosuggest&q=$query").text)
+        return response?.items?.first()?.results?.map { SearchResponse(Jsoup.parse(it.title).text(), it.permalink!!, it.thumbnail, null, null, name) }!! }
+
+
+    override suspend fun load(url: String): LoadResponse? {
+        val response = app.get(url)
+        val document = Jsoup.parse(response.text)
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/quicknovel/providers/MtlNovelsProvider.kt]: you dont need to write Jsoup.parse everywhere, just write `val document = app.get(url).document`

File: app/src/main/java/com/lagradost/quicknovel/providers/MtlNovelsProvider.kt [hunk 3]
```diff
@@ -0,0 +1,170 @@
+package com.lagradost.quicknovel.providers
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.lagradost.quicknovel.*
+import com.lagradost.quicknovel.MainActivity.Companion.app
+import org.jsoup.Jsoup
+
+class mtlNovelProvider : MainAPI() {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/quicknovel/providers/MtlNovelsProvider.kt]: mtlNovelProvider -> MtlNovelProvider

File: app/src/main/java/com/lagradost/quicknovel/providers/EfremnetProvider.kt [hunk 4]
```diff
@@ -0,0 +1,1502 @@
...
+        Pair("Americana", "americana"),
+        Pair("American South", "american-south"),
+        Pair("American Revolutionary War", "american-revolutionary-war"),
+        Pair("American Libertarianism", "american-libertarianism"),
+        Pair("Amateur Sleuth", "amateur-sleuth"),
+        Pair("All Ages", "all-ages"),
+        Pair("Alexander The Great", "alexander-the-great"),
+        Pair("African Literature", "african-literature"),
+        Pair("Adventure Thriller", "adventure-thriller"),
+    )
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/quicknovel/providers/EfremnetProvider.kt]: with this many providers, add `.sortedBy { it.first }` to make it sort alphabetically

File: app/src/main/java/com/lagradost/quicknovel/providers/EfremnetProvider.kt (hunk 1 above)
Comment (ghost) [reply app/src/main/java/com/lagradost/quicknovel/providers/EfremnetProvider.kt:1471]: It only works with it, I tried removing it and figuring out a fix

**Pull request merged**